### PR TITLE
Move JavaScript to separate file

### PR DESCRIPTION
**Pull request by technetium**
By creating a more general JavaScript code and putting it in a separate file, it can be used for all collections. This makes the code more DRY.

### CODE REVIEW

File: reference/forms/types/collection.rst [hunk 1]
```diff
@@ -158,6 +158,7 @@ things are even easier because the ``data-prototype`` attribute is rendered
 automatically for you (with a slight difference - see note below) and all
 you need is the JavaScript:
 
+
```
Comment (HeahDude) [inline reference/forms/types/collection.rst]: Should be reverted.

File: reference/forms/types/collection.rst [hunk 2]
```diff
@@ -176,35 +178,43 @@ you need is the JavaScript:
...
+    .. code-block:: javascript
+
+        // add-collection-widget.js
+        jQuery(document).ready(function () {
+            jQuery('.add-another-collection-widget').click(function (e) {
+                e.preventDefault();
+                var list = jQuery(jQuery(this).attr('data-list'));
+                // Try to find the counter of the list
+                var counter = list.data('widget-counter');
```
Comment (HeahDude) [inline reference/forms/types/collection.rst]: `var counter = list.data('widget-counter') | list.children().length;`

**Review by HeahDude — APPROVED** [review]
Looks like a good idea, thanks!

**Comment by technetium** [comment]
Let's see if it's help if I try to merge to 2.8
The link for platform.sh still goes to a 404 page:
No route found for pr-8205-mqerhba-6qmocelev2lwe.eu.platform.sh

**Comment by wouterj** [comment]
I think we shouldn't be using a configuration block here, but instead show the JS file in a seperate code block. I think it's important that the JS code is visible on the screen at first sight.

**Review by javiereguiluz — APPROVED** [review]

**Comment by javiereguiluz** [comment]
@technetium thanks for improving this code and congrats on your first Symfony Docs contribution!

**Pull request merged**